absl/time/internal/cctz/src/time_zone_lookup.cc

```diff
@@ -138,6 +138,43 @@ std::string win32_local_time_zone(const HMODULE combase) {
   calendar->Release();
   return result;
 }
+
+static std::string get_once_win32_local_time_zone() {
+  // Use the WinRT Calendar class to get the local time zone. This feature is
+  // available on Windows 10 and later. The library is dynamically linked to
+  // maintain binary compatibility with Windows XP - Windows 7. On Windows 8,
+  // The combase.dll API functions are available but the RoActivateInstance
+  // call will fail for the Calendar class.
+  std::string winrt_tz;
+  const HMODULE combase =
+      LoadLibraryEx(_T("combase.dll"), nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
+  if (combase) {
+    const auto ro_initialize = reinterpret_cast<decltype(&::RoInitialize)>(
+        GetProcAddress(combase, "RoInitialize"));
+    const auto ro_uninitialize = reinterpret_cast<decltype(&::RoUninitialize)>(
+        GetProcAddress(combase, "RoUninitialize"));
+    if (ro_initialize && ro_uninitialize) {
+      const HRESULT hr = ro_initialize(RO_INIT_MULTITHREADED);
+      // RPC_E_CHANGED_MODE means that a previous RoInitialize call specified
+      // a different concurrency model. The WinRT runtime is initialized and
+      // should work for our purpose here, but we should *not* call
+      // RoUninitialize because it's a failure.
+      if (SUCCEEDED(hr) || hr == RPC_E_CHANGED_MODE) {
+        winrt_tz = win32_local_time_zone(combase);
+        if (SUCCEEDED(hr)) {
+          ro_uninitialize();
+        }
+      }
+    }
+    FreeLibrary(combase);
+  }
+  return winrt_tz;
+}
+
+static std::string& cached_win32_local_time_zone() {
+  static std::string cached_time_zone = get_once_win32_local_time_zone();
```

Comment:
Note that this is not safe for global destruction. See https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables.

Reply:
Yes that is what I would do, but I was just pointing that out. I think the approach here is wrong anyway. See above.